DuckDuckGoTests/BrokenSitePromptLimiterTests.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  BrokenSitePromptLimiterTests.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import BrowserServicesKit
+@testable import DuckDuckGo
+@testable import Core
+
+final class BrokenSitePromptLimiterTests: XCTestCase {
+
+    let configManager = PrivacyConfigurationManagerMock()
+    var brokenSiteLimiter: BrokenSitePromptLimiter!
+
+    override func setUp() {
+        super.setUp()
+
+        setupUserDefault(with: #file)
+        UserDefaults.app.removeObject(forKey: UserDefaultsWrapper<String>.Key.lastBrokenSiteToastShownDate.rawValue)
+        UserDefaults.app.removeObject(forKey: UserDefaultsWrapper<String>.Key.toastDismissStreakCounter.rawValue)
+
+        (configManager.privacyConfig as? PrivacyConfigurationMock)?.enabledFeaturesForVersions[.brokenSitePrompt] = [AppVersionProvider().appVersion() ?? ""]
```

Comment:
relying on user defaults unnecessarily complicates the code in tests,
imho we shouldn't care what storage we use in the client, tests should operate on in memory variable that is being passed to your functions in your limiter, please see how I did UserBehaviorMonitor and UserBehaviorMonitorTests, there is similar use case.

in short there is handleAction that takes date: Date as argument, which is always provided in client through default argument like this date: Date = Date()

but in tests I can provide any date

also there is protocol: UserBehaviorStoring that defines the values that are stored.
in client there is Store that conforms to this protocol and all the values are being stored in user defaults, but in tests these variables are only in memory.